components/layout/layout_impl.rs

```diff
@@ -1049,20 +1049,29 @@ impl LayoutThread {
         self.epoch.set(epoch);
         stacking_context_tree.compositor_info.epoch = epoch.into();
 
+        let mut lcp_candidate = None;
```

Comment:
This pattern looks weird to me. It looks better if `lcp_candidate` is part of the output of `DisplayListBuilder::build`.